clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java

```
@@ -61,6 +59,17 @@ protected String getServiceName() {
   @Override
   protected void afterConnect() {
     mClient = new LineageMasterService.Client(mProtocol);
+    if (mServiceVersion == Constants.UNKNOWN_SERVICE_VERSION) {
+      try {
+        mServiceVersion = mClient.getServiceVersion();
+      } catch (TException e) {
+        throw new RuntimeException(e.getMessage());
+      }
+      if (!mCompatibleVersions.contains(mServiceVersion)) {
+        throw new RuntimeException(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(
```

Would it be better to throw a checked exception, since this is user facing?